.github/workflows/test.yml

```diff
@@ -1,31 +1,92 @@
-name: Test
+name: Tests
 
 # Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the master branch
+  # Triggers the workflow on push or pull request events
   push:
-    branches: [ tests ]
   pull_request:
-    branches: [ tests ]
 
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-    
-  # This workflow contains a single job called "tests"
-  tests:
-    # The type of runner that the job will run on
-    runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
+  windows:
+    strategy:
+      fail-fast: false
+      matrix:
+        python: [python38, python39]
+        rfVersion: [3.1.2, 3.2, 3.2.1, 3.2.2]
+    runs-on: windows-latest
+    name: Windows (${{ matrix.python }}, robotframework-${{ matrix.rfVersion }})
+    defaults:
+      run:
+        shell: wsl-bash {0}
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
+      - uses: Vampire/setup-wsl@v1
+        with:
+          distribution: Ubuntu-20.04
+          additional-packages:
+            dos2unix
+      - name: Install Nix
+        run: |
+            # Nix installer does not support root installation we need a user
+            useradd test -m
+
+            # Create /nix folder with proper permissions
+            # This is required because otherwise installer gets stuck most likely at running sudo
+            install -d -m755 -o test -g test /nix
 
-      - uses: cachix/install-nix-action@v13
+            # Download Nix installer
+            sudo -iu test curl -o ./install-nix.sh -L https://nixos.org/nix/install
+
+            # Run installer as single user installation and we do not need a channel
+            sudo -iu test sh ./install-nix.sh --no-daemon --no-channel-add
+      - name: Run tests
+        run: |
+            # Change line endings
+            dos2unix default.nix
+            # Run the tests in project directory where default.nix resides
+            # We need login shell for that so that .profile is getting read
+            sudo -iu test sh -c "nix-build $PWD --argstr python ${{ matrix.python }} --argstr rfVersion ${{ matrix.rfVersion }}"
+            exit $(cat /home/test/result/exitCode)
+
+  linux:
+    strategy:
+      fail-fast: false
+      matrix:
+        python: [python38, python39]
+        rfVersion: [3.1.2, 3.2, 3.2.1, 3.2.2]
+    runs-on: ubuntu-latest
+    name: Linux (${{ matrix.python }}, robotframework-${{ matrix.rfVersion }})
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Nix
+        uses: cachix/install-nix-action@v13
         with:
           nix_path: nixpkgs=channel:nixos-unstable
```

Comment:
I would appreciate explaining short comment why are we using unstable instead of stable

Reply:
I think it can even run without this option, since the default.nix and test.nix overrides it to `release-21.05`. Will try without it...